src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,169 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DOUBLING_FACTOR = 2;
+    private Node<K,V>[] table;
+    private int threshold;
+    private int size;
+
+    @SuppressWarnings({"unchecked"})
+    public MyHashMap() {
+        this.table = (Node<K,V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        this.size = 0;
+        this.threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            threshold *= DOUBLING_FACTOR;
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        int index = (table.length - 1) & newNode.hash;
+        if (table[index] == null) {
+            addNode(index, newNode);
+        } else {
+            Node<K, V> current = table[index];
+            while (current.next != null) {
+                if (areKeysEqual(current.key, newNode.key)) {
+                    changeValue(current, newNode);
+                    return;
+                } else {
+                    current = current.next;
+                }
+            }
+            if (areKeysEqual(current.key, newNode.key)) {
+                changeValue(current, newNode);
+                return;
+            }
+            appendNode(current, newNode);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = (table.length - 1) & hash(key);
+        if (index < table.length && table[index] != null) {
+            Node<K, V> current = table[index];
+            while (current != null) {
+                if (current.hash == hash(key)
+                        && Objects.equals(current.key, key)) {
+                    return current.value;
+                } else {
+                    current = current.next;
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public String toString() {
+        return "table=" + Arrays.toString(table);
+    }
+
+    @SuppressWarnings({"unchecked"})
+    private void resize() {
+        Node<K,V>[] oldTable = table;
+        table = (Node<K,V>[]) new Node[oldTable.length * DOUBLING_FACTOR];
+        size = 0;
+        transfer(oldTable);
+    }
+
+    private void addNode(int index, Node<K, V> newNode) {
+        table[index] = newNode;
+        size++;
+    }
+
+    private void appendNode(Node<K, V> current, Node<K, V> newNode) {
+        current.next = newNode;
+        size++;
+    }
+
+    private boolean areKeysEqual(K currentKey, K newKey) {
```

Comment:
Using public access modifiers everywhere is not a good practice. The method 'areKeysEqual' is only used inside this class, so it should be private to follow the encapsulation principle.